### PR TITLE
clipman: add module

### PR DESCRIPTION
**Pull request by jwygoda**
### Description

Add support for [clipman](https://github.com/yory8/clipman), a simple clipboard manager for Wayland.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/services/clipman.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+let
+
+  cfg = config.services.clipman;
+
+in {
+  meta.maintainers = [ maintainers.jwygoda ];
+
+  options.services.clipman = {
+    enable = mkEnableOption "clipman, a simple clipboard manager for wayland";
```
Comment (rycee) [inline modules/services/clipman.nix:13]:
```suggestion
    enable = mkEnableOption "clipman, a simple clipboard manager for Wayland";
```

File: modules/services/clipman.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.clipman;
+      defaultText = "pkgs.clipman";
+      description = "clipman derivation to use.";
+    };
+  };
```
Comment (rycee) [inline modules/services/clipman.nix:21]: Should be possible to simplify to
```suggestion
    package = mkPackageOption pkgs "clipman" { };
```

**Comment by rycee** [comment]
Thanks! Added some comments. Also, I believe you will need to add a reference to the module in `modules/modules.nix`. You can verify this by adding a test case 🙂

**Comment by jwygoda** [comment]
Thanks a lot for the review, @rycee! I've added a test case.

File: modules/services/clipman.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+in {
+  meta.maintainers = [ maintainers.jwygoda ];
+
+  options.services.clipman = {
+    enable = mkEnableOption "clipman, a simple clipboard manager for Wayland";
+
+    package = mkPackageOption pkgs "clipman" { };
+
+    systemd_target = mkOption {
```
Comment (rycee) [inline modules/services/clipman.nix:17]: Option names should, whenever possible, use camel case:

```suggestion
    systemdTarget = mkOption {
```

**Comment by teto** [comment]
is this ready? if Yes I can probably test tomorrow

**Comment by jwygoda** [comment]
Yes, it's ready. Thanks for the review!

**Comment by rycee** [comment]
Made a few minor edits and merged to master. Thanks for the contribution!

**Pull request merged**